packages/realm/src/Realm.ts

```diff
@@ -149,25 +150,25 @@ export class Realm {
    * @param loggerCallback - The callback invoked by the logger. The default callback uses `console.log`, `console.warn` and `console.error`, depending on the level of the message.
    * @note The logger callback needs to be setup before opening the first Realm.
    * @since 12.0.0
-   * @deprecated Pass a callback taking a single object argument instead.
    * @example
-   * Realm.setLogger((level, message) => {
-   *   console.log(`[${level}] ${message}`);
+   * Realm.setLogger(({ category, level, message }) => {
+   *   console.log(`[${category} - ${level}] ${message}`);
    * });
    */
-  static setLogger(loggerCallback: LoggerCallback1): void;
+  static setLogger(loggerCallback: LoggerCallback2): void;
```

Comment:
I'm moving this up to improve DX by hitting this as the default overload.

![Screenshot 2024-03-05 at 15 24 52](https://github.com/realm/realm-js/assets/1243959/1db8301e-6500-4927-b1c1-b1fdf5d7a10c)

Reply:
But that could be considered breaking 😬

![Screenshot_2024-03-05-16-54-38-91_40deb401b9ffe8e1df2f1cc5ba480b12~2](https://github.com/realm/realm-js/assets/1243959/d63bc421-8363-4396-abb2-bfe45b82271f)